autowarefoundation/autoware.universe
# feat(behavior_path_planner): output multiple candidate paths

Signed-off-by: Fumiya Watanabe <rej55.g@gmail.com>

## Description

In the future, multiple modules in behavior_path_planner will be run.
To handle this, in this change, I change behavior_path_planner to output multiple candidate paths.

In this change, the name space of candidate path topics is also changed.

Before:
`/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/output/path_candidate`

After:
`/planning/path_candidate/**`

Furthermore, it also outputs multiple RTC status.
So, it becomes able to receive RTC commands if the modules request.

## Related links



## Tests performed

Tested in the planning simulator.
See also https://github.com/autowarefoundation/autoware_launch/pull/125

- Lane change & Avoidance
![Screenshot from 2022-12-11 22-08-36](https://user-images.githubusercontent.com/10190493/206906604-8f657546-48d6-4b60-8a45-dcadba4e393e.png)

- Pull out & Avoidance
![Screenshot from 2022-12-11 22-13-53](https://user-images.githubusercontent.com/10190493/206906640-c907a8b6-89ea-40a8-8f13-bc0315768696.png)

- Pull over & Avoidance
![Screenshot from 2022-12-11 22-22-45](https://user-images.githubusercontent.com/10190493/206906651-2ce99b6a-275f-49cd-a0a4-aed9e955bf48.png)

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
